subsys/bluetooth/host/hci_core.c

```diff
@@ -2502,6 +2502,35 @@ void bt_hci_le_transmit_power_report(struct net_buf *buf)
 }
 #endif /* CONFIG_BT_TRANSMIT_POWER_CONTROL */
 
+#if defined(CONFIG_BT_PATH_LOSS_MONITORING)
+void bt_hci_le_path_loss_threshold_event(struct net_buf *buf)
+{
+	struct bt_hci_evt_le_path_loss_threshold *evt;
+	struct bt_conn_le_path_loss_threshold_report report;
+	struct bt_conn *conn;
+
```

Comment:
I think we should code defensively against the controller. 

```suggestion

	if (buf->len < sizeof(*evt)) {
		LOG_ERR("Too short bt_hci_evt_le_path_loss_threshold");
		return;
	}
```

We should also maybe check that `evt->zone_entered` and `evt->current_path_loss` is one of the expected values before we pass it on to the application. @jhedberg What do you think about this? The application should also be defensively coded, so it may be fair to just pass on the value as is.

Reply:
Is this doing the same thing in the generic event handling code? https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/hci_core.c#L158

Reply:
@sean-madigan You are right. Please ignore that part of the suggestion.

Reply:
Happy to add a check that `evt->zone_entered` is a sane value. I think any value of `evt->current_path_loss` _could_ be valid

Reply:
I tend to agree with @alwa-nordic that we should be defensive, even against the controller. 

This is something that we today do quite poorly, as we generally tend to trust the controller (which we kind of need to do in order to work in the first place). 

In any case, I'd be for checking that the value is in a valid range